### PR TITLE
Tests for tar_interpreter.go -> TarInterpreter -> Interpret

### PR DESCRIPTION
**Pull request by dimastark**

### CODE REVIEW

File: test/tar_interpreter_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	"bytes"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wal-g/wal-g/internal"
+)
+
+type testInterpretOptions struct {
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: What is the reason for creating this struct? it doesn't simplify the code for reading and hide dependencies.
Use KISS principle here.

File: test/tar_interpreter_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+func testInterpret(t *testing.T, options testInterpretOptions) (os.FileInfo, os.FileInfo) {
+	tarInterpreter := &internal.FileTarInterpreter{
+		DBDataDirectory: options.DBDataDirectory,
+	}
+
+	err := options.Create(options.Name)
+
+	assert.NoError(t, err)
+
+	defer options.Delete(options.Name)
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: It looks like an unhandled error. Try to replace it with something similar to this https://pastebin.com/7xAFJtpX

File: test/tar_interpreter_test.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		&bytes.Buffer{},
+		&tar.Header{
+			Name:     options.Name,
+			Typeflag: options.Typeflag,
+		},
+	)
+
+	assert.NoError(t, err)
+
+	defer options.Delete(path.Join(options.DBDataDirectory, options.Name))
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: It looks like an unhandled error

File: test/tar_interpreter_test.go (hunk 1 above)
Comment (g0djan) [reply test/tar_interpreter_test.go]: Passing keyword arguments is redundant here.

File: test/tar_interpreter_test.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		DBDataDirectory: dbDataDirectory,
+
+		Name:     "test_dir",
+		Typeflag: tar.TypeDir,
+
+		Create: mkdirAll,
+		Delete: os.RemoveAll,
+	})
+
+	defer os.RemoveAll(dbDataDirectory)
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: Unhandled error

File: test/tar_interpreter_test.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+	DBDataDirectory string
+
+	Name     string
+	Typeflag byte
+
+	Create func(string) error
+	Delete func(string) error
+}
+
+func testInterpret(t *testing.T, options testInterpretOptions) (os.FileInfo, os.FileInfo) {
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: You don't cover cases when an error occurs.

Comment (g0djan) [reply test/tar_interpreter_test.go]: You can use assert.IsType for it.

File: test/tar_interpreter_test.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	assert.NoError(t, err)
+
+	return srcFileInfo, dstFileInfo
+}
+
+func mkdirAll(path string) error {
+	return os.MkdirAll(path, 0766)
+}
+
+func justCreate(path string) error {
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: Rename it to createFile and comment that it's needed for unification returning type with createDir.

File: test/tar_interpreter_test.go [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+	assert.NoError(t, err)
+
+	dstFileInfo, err := os.Lstat(path.Join(options.DBDataDirectory, options.Name))
+
+	assert.NoError(t, err)
+
+	return srcFileInfo, dstFileInfo
+}
+
+func mkdirAll(path string) error {
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: Rename to createDir for a unique name with createFile

File: test/tar_interpreter_test.go (hunk 6 above)
Comment (dimastark) [reply test/tar_interpreter_test.go]: I think a comment is not needed, the purpose of the function becomes obvious if you look at the place of use.

Comment (dimastark) [reply test/tar_interpreter_test.go]: I think a comment is not needed, the purpose of the function becomes obvious if you look at the place of use.

File: test/tar_interpreter_test.go (hunk 5 above)
Comment (dimastark) [reply test/tar_interpreter_test.go]: I write this function only for **happy path** cases. I'll deal with **bad path** later.

File: test/tar_interpreter_test.go [hunk 8]
```diff
@@ -11,48 +11,38 @@ import (
...
-
-	Name     string
-	Typeflag byte
-
-	Create func(string) error
-	Delete func(string) error
-}
-
-func testInterpret(t *testing.T, options testInterpretOptions) (os.FileInfo, os.FileInfo) {
+func testInterpret(t *testing.T, dbDataDirectory, name string, typeflag byte, create, delete func(string) error) (os.FileInfo, os.FileInfo) {
```
Comment (g0djan) [inline test/tar_interpreter_test.go]: Break this line into several

Comment (g0djan) [reply test/tar_interpreter_test.go]: Nice work. Fix it and I'll merge it.

File: test/tar_interpreter_test.go (hunk 5 above)
Comment (g0djan) [reply test/tar_interpreter_test.go]: Ok, thanks. We will wait for your next PR :)

**Pull request merged**